src/registry.js

```diff
@@ -117,8 +117,7 @@ class CustomAttributeRegistry {
       }
     }
     // Attribute was removed
-    else if(newVal == null && !!inst.value) {
```

Comment:
Example: if a custom attribute is `some-attribute=""`, and then it gets removed, the `newVal` will be `null`, and the current `inst.value` will `""` which is falsy, which then causes the conditional to incorrectly skip calling `disconnectedCallback`.